gravitee-apim-console-webui/src/management/configuration/metadata/environment-metadata.component.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { HarnessLoader } from '@angular/cdk/testing';
+import { HttpTestingController } from '@angular/common/http/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { MatIconTestingModule } from '@angular/material/icon/testing';
+import { InteractivityChecker } from '@angular/cdk/a11y';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { GioConfirmDialogHarness } from '@gravitee/ui-particles-angular';
+
+import { EnvironmentMetadataComponent } from './environment-metadata.component';
+import { EnvironmentMetadataModule } from './environment-metadata.module';
+
+import { User } from '../../../entities/user';
+import { CurrentUserService, UIRouterStateParams } from '../../../ajs-upgraded-providers';
+import { CONSTANTS_TESTING, GioHttpTestingModule } from '../../../shared/testing';
+import { GioMetadataHarness } from '../../../components/gio-metadata/gio-metadata.harness';
+import { GioMetadataDialogHarness } from '../../../components/gio-metadata/dialog/gio-metadata-dialog.harness';
+import { fakeMetadata } from '../../../entities/metadata/metadata.fixture';
+import { Metadata } from '../../../entities/metadata/metadata';
+
+describe('EnvironmentMetadataComponent', () => {
+  let fixture: ComponentFixture<EnvironmentMetadataComponent>;
+  let loader: HarnessLoader;
+  let rootLoader: HarnessLoader;
+  let httpTestingController: HttpTestingController;
+  const API_ID = 'my-api';
+
+  const init = async () => {
+    const currentUser = new User();
+    currentUser.userPermissions = ['environment-metadata-u', 'environment-metadata-d', 'environment-metadata-c'];
+
+    await TestBed.configureTestingModule({
+      declarations: [EnvironmentMetadataComponent],
+      providers: [
+        { provide: UIRouterStateParams, useValue: { apiId: API_ID } },
+        { provide: CurrentUserService, useValue: { currentUser } },
+      ],
+      imports: [NoopAnimationsModule, GioHttpTestingModule, EnvironmentMetadataModule, MatIconTestingModule],
+    })
+      .overrideProvider(InteractivityChecker, {
+        useValue: {
+          isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+          isTabbable: () => true, // Allows to choose a day in the calendar
+        },
+      })
+      .compileComponents();
+
+    fixture = TestBed.createComponent(EnvironmentMetadataComponent);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    loader = TestbedHarnessEnvironment.loader(fixture);
+    rootLoader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+    fixture.detectChanges();
+  };
+
+  beforeEach(async () => await init());
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    httpTestingController.verify();
+  });
+
+  it('should load metadata list', async () => {
+    expectMetadataList();
+    const gioMetadata = await loader.getHarness(GioMetadataHarness);
```

Comment:
Just a note: here you're using a Harness made with the metadata component, which makes it easy to test here. ;)

Reply:
Okay, I'll look into this :)